R/mz_summary.R

```diff
@@ -0,0 +1,135 @@
+
+#' @name mz_summary
+#'
+#' @aliases mz_summary
+#'
+#' @title Create a summary from adjacency list containing mass-differences with possible filter 
+#'
+#' @description
+#' The function `mz_summary` creates a summary from the `AdjacencyMatrix`, containing mass-differences.
+#' Individual mass differences are count over all features. The input may be 
+#' `AdjacencyMatrix` originating from the function `structural`,
+#' or `combine`. The parameter `filtered` will define if data will be 
+#' filtered above a certain threshold or not. 
+#' 
+#' @param x
+#' `AdjacencyMatrix`, a formal class of `AdjacencyMatrix` containing the mass-differences, that have previously 
+#' been generated by the function `structural`or `combine`
+#' 
+#' @param filter
+#' `number`/`FALSE`, leave empty or set to `FALSE` if unfiltered data are 
+#' required. Select a `number` as a threshold where mz differences were filtered.
+#' May be useful at visualization for big data. 
+#' 
+#' @details
+#' Summarizes the adjacency matrices containing mass difference values, 
+#' i.e. either adjacency list from `structural`or `combine` may be used.
+#' Also the plots will be displayed. 
+#' The default is filter = F, so the unfiltered summary will be returned. 
+#' If filter is set to a `number`, e.g. 1000 only mz differences above 
+#' this threshold will be displayed. 
+#' The function can be applied for adjacency lists from `structural` and `combine`
+#' 
+#' @return 
+#' `data.frame` containing the numbers of present mz differences and
+#' corresponding name. Also a plot will be displayed.
+#'
+#' @author Liesa Salzer, \email{liesa.salzer@@helmholtz-muenchen.de}
+#'
+#' @examples
+#' data("x_test", package = "MetNet")
+#' transformation <- rbind(
+#'     c("Monosaccharide (-H2O)", "C6H10O5", "162.0528234315"),
+#'     c("Disaccharide (-H2O)", "C12H20O11", "340.1005614851"),
+#'     c("Trisaccharide (-H2O)", "C18H30O15", "486.1584702945"))
+#' transformation <- data.frame(group = transformation[, 1],
+#'                                 formula = transformation[, 2],
+#'                                 mass = as.numeric(transformation[, 3]))
+#' am_struct <- structural(x_test, transformation, ppm = 5, directed = TRUE)
+#' mz_summary(am_struct)
+#'
+#'
+#'@export
+#'
+#'@importFrom magrittr %>%
+mz_summary <- function(x, filter = F, ...){
+  
+  if (!"AdjacencyMatrix" %in% class(x)) 
+    stop("x is not class AdjacencyMatrix")
+  
+  x_df <- as.data.frame(x)
+
+  if (!"mass_difference" %in% colnames(x_df)) 
```

Comment:
please check if `validObject(x)` will do the same job. THen we don't have to deal with the colnames of `x`/`x_df`

Reply:
I think `validObject(x)` would not work in this case but I can instead check if `type(x)` is `"combine"` or `"structural"` (because these contain mass-differences)